demos/common/monitors/presenter.cpp

```diff
@@ -48,7 +48,8 @@ Presenter::Presenter(const std::string& keys, int yPos, cv::Size graphSize, std:
 
 void Presenter::addRemoveMonitor(MonitorType monitor) {
     int sampleStep = std::max(1, static_cast<int>(graphSize.width / historySize));
```

Comment:
Doesn't the `sampleStep` need to be changed too? E.g. if `historySize` is 2, `sampleStep` ought to be equal to `graphSize.width` (since there would be one point on the very left and one point on the very right).

Reply:
Sorry, I don't understand your proposal.
Given `historySize == 2`, `sampleStep` becomes `graphSize.width/2` or `1` if 2 >= graphSize.width. So the only case when `sampleStep == graphSize.width` is when `graphSize.width == 1`. In that case `updatedHistorySize` will be 2, which I think is fine.

Reply:
> Given historySize == 2, sampleStep becomes graphSize.width/2 or 1 if 2 >= graphSize.width.

Yeah, but isn't this wrong? `sampleStep` is the distance between points in the graph, right? So if there are only two points, then the left point will be on the very left, the right point will be on the very right, so the distance between them should be the entire width of the graph.

Reply:
updated